src/main/scala/org/broadinstitute/hail/expr/Type.scala

```diff
@@ -328,6 +327,11 @@ case class TDict(elementType: Type) extends Type {
 
   override def genValue: Gen[Annotation] =
     Gen.buildableOf2[Map, String, Annotation](Gen.zip(arbitrary[String], elementType.genValue))
+
+  override def valuesSimilar(a1: Annotation, a2: Annotation, tolerance: Double): Boolean =
+    a1 == a2 || (a1 != null && a2 != null
+      && a1.asInstanceOf[Map[String, _]].zip(a2.asInstanceOf[Map[String, _]])
+      .forall { case ((k1, v1), (k2, v2)) => k1 == k2 && elementType.valuesSimilar(v1, v2, tolerance) })
```

Comment:
I'd like to push back to leave a1 == a2, because checking equality is going to be way faster, and usually sufficient -- we only do 2 passes when a) there are floating point mismatches, or b) when the annotations aren't the same, which is a test failure case.